drivers/epson/projector/esc_vp21.cr

```diff
@@ -0,0 +1,274 @@
+require "digest/md5"
+require "placeos-driver/interface/muteable"
+require "placeos-driver/interface/powerable"
+require "placeos-driver/interface/switchable"
+
+module Epson; end
+
+module Epson::Projector; end
+
+# Documentation: documentation link
+
+class Epson::Projector::EscVp21 < PlaceOS::Driver
+  include Interface::Powerable
+  include Interface::Muteable
+
+  enum Inputs
+    HDMI
+    HDBaseT
+  end
+
+  # Discovery Information
+  tcp_port 1024
+  descriptive_name "Epson Projectors"
+  generic_name :Display
+
+  def on_load
+    transport.tokenizer = Tokenizer.new("\r\n")
+
+    self[:volume_min] = 0
+    self[:volume_max] = 255
+
+    self[:power] = 0
```

Comment:
Best to set any initial state based on a device query, rather than explicit value.

Reply:
Could you clarify what you mean by a 'device query'?

Reply:
Sending a command to the device that returns the device state. For example, it looks like you already have `power?` and `input?` implemented for this.

The reason for this is if a driver is stopped from backoffice, when it restarts you don't want to generate state change events _unless_ the state has actually changed. A nice pattern to follow is use the `connected` method to query any state that the driver is exposing. That way if the device drops offline or the driver stops when connectivity is re-established the driver always shows the correct device state.